ClassFieldTheory/Cohomology/Functors/Corestriction.lean

```diff
@@ -127,11 +128,50 @@ def cores₁_obj [DecidableEq G] (M : Rep R G) :
     convert comp_zero -- cancel first functor
     exact (mapShortComplex₃ (up_shortExact M) (rfl : 0 + 1 = 1)).zero
 
+@[reassoc]
+lemma commSq_cores₁ [DecidableEq G] (M : Rep R G) :
+  δ (up_shortExact_res M S.subtype) 0 1 rfl ≫ cores₁_obj (S := S) M =
+    (cores₀ (S := S)).app _ ≫ δ (up_shortExact M) 0 1 rfl :=
+  haveI : Epi (mapShortComplex₃ (up_shortExact_res M S.subtype) (rfl : 0 + 1 = 1)).g :=
+    epi_δ_up_zero_res (R := R) (φ := S.subtype) M S.subtype_injective
+  (mapShortComplex₃_exact (up_shortExact_res M S.subtype) (rfl : 0 + 1 = 1)).g_desc _ _
+
+theorem δ_naturality {X1 X2 : ShortComplex (Rep R G)} (hX1 : X1.ShortExact)
```

Comment:
This has nothing to do with cores, right? Can you put it in `ClassFieldTheory.Mathlib.RepresentationTheory.Homological.GroupCohomology.LongExactSequence`? You'll have to import the Mathlib file (for the statement) and also `Mathlib.Algebra.Homology.HomologySequenceLemmas` (for the proof). Or even better, put it in mathlib?